### ISSUE TITLE
Problem with some files

### ISSUE THREAD

**Issue by antoninnoel** [philiphardy/ID3Edit#1]
It seems that with some files, the version byte is not the right one, in my song : the version byte is 3 against 2 on your header array.
What should i do so ?

**Comment by philiphardy** [comment]
Hey there. Yes, it is only compatible with version 2 tags for now. I just
haven't had time to include compatibility for version 3 yet.

If you'd like to contribute that would be awesome.
The specifications for the version 3 can be found here:
http://id3.org/id3v2.3.0

**Comment by antoninnoel** [comment]
Id like to, but i'm not that good with byte manipulation ... Sorry !

**Comment by philiphardy** [comment]
No worries. I'll try and get that up soon

**Comment by flocked** [comment]
Any news about that? :) I'm also not good in byte manipulation else I would help.

**Comment by philiphardy** [comment]
Sorry, I've just been swamped with school projects. I promise I will get to it come summer. (6 weeks)

**Comment by ryanfrancesconi** [comment]
I'm working on an update of your code for v3. I'll send it over if I can get it more polished up soon. I'm working with ID3 in Wave files though right now, so perhaps it is a little different.

Thanks for the groundwork, Phillip.

Ryan

**Comment by ryanfrancesconi** [comment]
If you only need to parse MP3 metadata then the AVFoundation classes actually will do that for you fine:
AudioFileGetProperty( yourAudioFileID,
                                                     kAudioFilePropertyInfoDictionary,
                                                     &propSize,
                                                     &metadata

I found that this method doesn't work properly with WAVE files however so I needed to dig deeper. Note that these Objective C methods are somewhat messy to interact with Swift so using the [EZAudioFile](https://github.com/syedhali/EZAudio/blob/master/EZAudio/EZAudioFile.m) class is the easiest way to go if you want to pull the metadata. That returns a nice neat NSDictionary. I only mention this here as the built in stuff will read the various ID3 versions fine. (For a mp3 etc)
